### PR TITLE
Provide access_token in header instead of URI

### LINKED ISSUE #48: Changes in GitHub Authorisation
GitHub will disallow API authentication through query parameters (see [blog post](https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param/) in the near future. Authorisation must use approriate headers instead.

In case of `client_id:client_secret`, the credentials are handled correctly already as user/password pair.
in case of access token authorisation, the token need to be transferred as a header instead of as a query parameter.

### PR DESCRIPTION
**Pull request by nibra**
Pull Request for Issue #48

### Summary of Changes

Adds the `authHeader()` method to \Joomla\Github\AbstractGithubObject.
Changes unit tests to ensure the required treatment of header information.

### Testing Instructions

Verify that the changes in the unit tests will test the right things.
Run PHPUnit.

### Documentation Changes Required

API documentation will be updated automatically.

### CODE REVIEW

File: src/AbstractGithubObject.php [hunk 1]
```diff
@@ -9,7 +9,7 @@
 namespace Joomla\Github;
 
 use Joomla\Http\Exception\UnexpectedResponseException;
-use Joomla\Http\Http as BaseHttp;
+use Joomla\Http\Http;
```
Comment (wilsonge) [inline src/AbstractGithubObject.php:12]: Please revert this change - there is already a class in this package called Http https://github.com/joomla-framework/github-api/blob/f58a08de8c8238712a67406356e78c8ac13ceb74/src/Http.php

File: Tests/GithubObjectTest.php [hunk 2]
```diff
@@ -109,9 +146,16 @@ public function testFetchUrlToken()
...
+			'https://api.github.com/gists',
 			$this->object->fetchUrl('/gists', 0, 0),
-			$this->equalTo('https://api.github.com/gists?access_token=MyTestToken')
+			'URL is not as expected.'
+		);
+
+		self::assertEquals(
+			array('Authorization' => 'token MyTestToken'),
+			$this->client->getOption('headers'),
+			'Token should bhe propagated as a header.'
```
Comment (wilsonge) [inline Tests/GithubObjectTest.php:158]:
```suggestion
			'Token should be propagated as a header.'
```

Comment (nibra) [reply Tests/GithubObjectTest.php]: Fixed, thank you!

File: src/AbstractGithubObject.php (hunk 1 above)
Comment (nibra) [reply src/AbstractGithubObject.php]: Fixed, thank you!

**Comment by HLeithner** [comment]
@nibra @zero-24 @wilsonge  can we merge this?

**Comment by nibra** [comment]
Yes. The "failing check" is caused by a composer problem in Travis, not by this PR.

**Comment by HLeithner** [comment]
@nibra can you update your branch to get drone foe tests plz

**Comment by nibra** [comment]
@HLeithner Done

**Comment by zero-24** [comment]
> @nibra @zero-24 @wilsonge can we merge this?

Seems this is awainting my feedback I'm fine with that PR. Would be great to get that merged and tag a new release

**Comment by zero-24** [comment]
@nibra can you please do a 1.7.1 release with the changes done here?

**Comment by nibra** [comment]
Done.

**Comment by zero-24** [comment]
Thanks
